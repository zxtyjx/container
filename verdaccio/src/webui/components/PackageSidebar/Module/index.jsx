import React from 'react';
import PropTypes from 'prop-types';

import classes from './style.scss';

export default function Module({title, description, children, className}) {
  return (
    <div className={`${classes.module} ${className}`}>
      <h2 className={classes.moduleTitle}>
        {title}
        {description && <span>{description}</span>}
      </h2>
      <div>
        {children}
      </div>
    </div>
  );
}

Module.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  children: PropTypes.any.isRequired,
  className: PropTypes.string
};
