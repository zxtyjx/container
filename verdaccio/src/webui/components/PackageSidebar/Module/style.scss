@import '../../../styles/variable';

.module {

  margin-bottom: 10px;

  .moduleTitle {
    display: flex;
    align-items: flex-end;
    font-size: 21px;
    margin: 0 0 10px;
    padding: 5px 0;
    font-weight: 600;
    border-bottom: 1px solid $border-color;

    span { // description
      font-size: 14px;
      color: $text-grey;
      margin-left: auto;
      font-weight: lighter;
    }
  }
}
