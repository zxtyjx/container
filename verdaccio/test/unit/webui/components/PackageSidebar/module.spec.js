/**
 * Module component
 */

import React from 'react';
import { shallow } from 'enzyme';
import Module from '../../../../../src/webui/components/PackageSidebar/Module/index';

console.error = jest.fn();

describe('<PackageSidebar /> : <Module />', () => {
  it('should error for required props', () => {
    shallow(<Module />);
    expect(console.error).toBeCalled();
  });
  it('should load module component', () => {
    const props = {
      title: 'Test title',
      description: 'Test description',
      className: 'module-component'
    };
    const wrapper = shallow(
      <Module {...props}>
        <p>test children</p>
      </Module>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
