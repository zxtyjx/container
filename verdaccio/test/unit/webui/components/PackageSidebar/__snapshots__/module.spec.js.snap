// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageSidebar /> : <Module /> should load module component 1`] = `"<div class=\\"module module-component\\"><h2 class=\\"moduleTitle\\">Test title<span>Test description</span></h2><div><p>test children</p></div></div>"`;
